lark/tree.py

```diff
@@ -52,9 +53,9 @@ class Tree(Generic[_Leaf_T]):
     """
 
     data: str
-    children: 'List[Union[_Leaf_T, Tree[_Leaf_T]]]'
+    children: 'List[Branch[_Leaf_T]]'
 
-    def __init__(self, data: str, children: 'List[Union[_Leaf_T, Tree[_Leaf_T]]]', meta: Optional[Meta]=None) -> None:
+    def __init__(self, data: str, children: 'List[Branch[_Leaf_T]]', meta: Optional[Meta]=None) -> None:
```

Comment:
Sorry if it's obvious, but why do we need to write `Branch[_Leaf_T]` and not just `Branch`, since it's already defined using `_Leaf_T`?

Reply:
Not a problem, it is a subtle point. At the top level `_Leaf_T` stand for some type, but it can still be anything. Once a variable containing a tree is defined as `Tree[Foo]`, then that tree's `_Leaf_T` is bound to `Foo`. Since `Branch` is defined at the top level, it can still be any type. Specifically using `Branch[_Leaf_T]` within the context of a `Tree`, means that the type has to match the bound `_Leaf_T` (in our example `Foo`).